src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -345,6 +338,30 @@ const StudentApplication = ({ userProfile, authentication }) => {
     }
   };
 
+  /**
+   * Callback for changes to off-campus program info
+   * @param {String} offCampusProgramValue The program that the applicant is doing an OC program for
+   * @param {Number} index The index of the applicant in the list
+   */
+  const handleOffCampusInputChange = (offCampusProgramValue, index) => {
+    if (index !== null && index >= 0) {
+      // Get the profile of the selected user
+      let newApplicant = {
+        ...applicants[index],
+        OffCampusProgram: offCampusProgramValue ?? applicants[index].OffCampusProgram,
+      };
+      // Error checking on the hallNameValue before modifying the newHallInfo object
+      setApplicants((prevApplicants) =>
+        // replace the element at index with the new hall info object
+        prevApplicants.map((prevApplicant, j) => (j === index ? newApplicant : prevApplicant)),
+      );
+    } else {
+      setSnackbarText('Something went wrong while trying to add this person. Please try again.');
```

Comment:
```suggestion
      setSnackbarText('Something went wrong while trying to change the off-campus program. Please try again.');
```

Reply:
@EjPlatzer feel free to commit this or modify it, since we do not have auth to commit directly into this branch